### PR TITLE
Our modern symphony

### PR DESCRIPTION
**Pull request by kazmasaurus**
This changes `Composer`s so they hold an array of child references, instead of just one.

Also includes:
- A shit ton of tests to make sure stuff is getting retained/released as expected
- A host application to run the tests in, (because otherwise the `UIViewController.presentViewController` API's don't work right, since there's no UIWindow to present to.)

### CODE REVIEW

File: Symphony/Composer.swift [hunk 1]
```diff
@@ -22,36 +22,43 @@ public protocol Composer: class {
     /// A reference to the currently presented composable.
     /// This is generally only necissary for memory management,
     /// and shouldn't need to be messed with.
-    var currentComposable: Composable? { get set }
```
Comment (jgrund01) [inline Symphony/Composer.swift:25]: Lets make this a set!

File: SymphonyTests/ComposerTests.swift [hunk 2]
```diff
@@ -7,20 +7,388 @@
...
+    }
+
+
+    func test_whenPresents_childAndPresentedAreBothRetained() {
+        weak var weakChildComposable: BasicComposable?
+        weak var weakPresentedComposable: BasicComposable?
+
+        autoreleasepool {
+            let childComposable = BasicComposable()
+            let presentedComposable = BasicComposable()
```
Comment (jgrund01) [inline SymphonyTests/ComposerTests.swift:138]: child -> shown

**Review by jgrund01 — CHANGES_REQUESTED** [review]
Looks awesome, nice job catching all the edge cases of this!  Just one rename we talked about for clarity

File: Symphony/Composer.swift (hunk 1 above)
Comment (kazmasaurus) [reply Symphony/Composer.swift:25]: So, I realized after I started trekking down this path that it's not as obvious a win as I was hoping. While I do think `Set` is the most appropriate data structure (`currentComposables` _should_ be unique and unordered), making `Composable: Equatable & Hashable` is non trivial (especially in Swift 2), so I'm voting we punt for the time being 😞

Comment (jgrund01) [reply Symphony/Composer.swift:25]: Sounds good, if it's non trivial and it was a "nice to have" lets punt till it becomes more of a need

**Review by jgrund01 — APPROVED** [review]
Looks great, names map more clearly to the functions!

**Comment by jgrund01** [comment]
👍 🔥

**Pull request merged**